src/trace/trace-context-service.ts

```diff
@@ -33,41 +33,62 @@ export class TraceContextService {
       return;
     }
     const traceContext = { ...this.rootTraceContext };
+
+    // Update the parent id to the active datadog span if available
     const datadogContext = this.tracerWrapper.traceContext();
     if (datadogContext) {
       logDebug(`set trace context from dd-trace with parent ${datadogContext.parentID}`);
       return datadogContext;
     }
 
+    // Update the parent id to the active X-Ray subsegment if available
     const xraySegment = this.getXraySegment();
     if (xraySegment === undefined) {
       logDebug("couldn't retrieve segment from xray");
     } else {
       const value = convertToAPMParentID(xraySegment.id);
       if (value !== undefined) {
-        logDebug(`set trace context from xray with parent ${value} from segment`);
+        logDebug(`set trace context from xray with parent ${value} from segment`, { xraySegment });
         traceContext.parentID = value;
       }
     }
 
     return traceContext;
   }
 
+  // Get the current trace headers to be propagated to the downstream calls,
+  // The parent id always points to the current active span.
   get currentTraceHeaders(): Partial<TraceHeaders> {
-    if (this.currentTraceContext === undefined) {
+    const traceContext = this.currentTraceContext;
+    if (traceContext === undefined) {
+      return {};
+    }
+    return {
+      [traceIDHeader]: traceContext.traceID,
+      [parentIDHeader]: traceContext.parentID,
+      [samplingPriorityHeader]: traceContext.sampleMode.toString(10),
+    };
+  }
+
+  // Get the trace headers from the root trace context.
+  get rootTraceHeaders(): Partial<TraceHeaders> {
+    const rootTraceContext = this.rootTraceContext;
```

Comment:
It looks like we set the root trace context to being the incoming Datadog trace headers. However, when hybrid tracing is enabled, we want to use the PARENT ID from X-Ray but the TRACE ID from Datadog. Is that accounted for here?

Reply:
Here is the relevant logic in Go: https://github.com/DataDog/datadog-lambda-go/blob/main/internal/trace/context.go#L45

Reply:
Yes, the root trace context is derived from X-Ray when there is not incoming Datadog trace headers.

Reply:
I think Nick is right about the desired behaviour. Otherwise the parent on the root span swaps between the incoming datadog span, (when it's available), and the root x-ray segment. This was an existing issue, so we don't have to fix it with this PR, but I think we would change the logic here: `https://github.com/DataDog/datadog-lambda-js/blob/main/src/trace/context.ts#L88`, to set the parentId to the x-ray when merge x-ray traces is enabled.